### PR TITLE
WSSQL-70 Provide String Wildcard support

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds 'LIKE' and 'NOT LIKE' operator support
- Maps LIKE op to ECL STD.Str.WildMatch function

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [x] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx please review.

File: esp/services/ws_sql/SQL2ECL/HPCCSQLTreeWalker.cpp [hunk 1]
```diff
@@ -324,6 +326,14 @@ ISQLExpression * HPCCSQLTreeWalker::expressionTreeWalker(pANTLR3_BASE_TREE exprA
                         paramList.append(*rightexp.getLink());
                     }
                 }
+                if (exptype == LIKE_SYM || exptype == NOT_LIKE)
+                {
+                    if (rightexp->getExpType() == Value_ExpressionType)
+                    {
+                        SQLValueExpression * valexp = static_cast<SQLValueExpression *>(rightexp.get());
+                        valexp->setIsWildCardPattern(true);
+                    }
```
Comment (wangkx) [inline esp/services/ws_sql/SQL2ECL/HPCCSQLTreeWalker.cpp]: Is it possible that you need an 'else' condition?

Comment (rpastrana) [inline esp/services/ws_sql/SQL2ECL/HPCCSQLTreeWalker.cpp]: Good question, in general an if should also account for the else. In this case however, the else would have already been taken care of by lines 312-328...
I could have not processed case LIKE_SYM and case NOT_LIKE on their own, but I didn't want to duplicate 312-315 in the new case...

**Comment by wangkx** [comment]
@rpastrana ok, looks fine.

**Pull request merged**